### PR TITLE
[rb] Adding method to send DevTools commands to Chrome and a helper method to enable downloads on headless Chrome.

### PR DESCRIPTION
**Pull request by pulkitsharma07**
For: https://github.com/SeleniumHQ/selenium/issues/5159

Files can now be downloaded in headless mode after setting the download directory's path using the `#download_path=` helper.

Working fine with Chrome 65 (chromedriver 2.36). For earlier versions of Chrome/chromedriver I was facing crashes when downloading files in headless mode.

- [X] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

---
This change is [Reviewable](https://reviewable.io/reviews/seleniumhq/selenium/5583)

### CODE REVIEW

File: rb/spec/integration/selenium/webdriver/chrome/driver_spec.rb [hunk 1]
```diff
@@ -43,6 +43,65 @@ module Chrome
...
+              expect(File).to exist(@downloaded_file_path)
+            end
+
+            it 'can download files in non-headless mode' do
+              driver.download_path = @download_dir
+              driver.get(@page_to_download_from)
+              driver.find_element(css: 'a').click
+
+              # Wait for file to download
+              5.times do
```
Comment (luke-hill) [inline rb/spec/integration/selenium/webdriver/chrome/driver_spec.rb]: This logic can be used using a wait.until command much cleaner, and with more control. If you wait for the temporary file not to be there (It's hard to check but I "think" I may have mentioned this before.

**Comment by lmtierney** [comment]
After talking with the chromedriver team, the `send_command` will be moved to a w3c-compliant protocol extension endpoint (https://w3c.github.io/webdriver/webdriver-spec.html#protocol-extensions). We may want to hold off on this PR until the change.

See the issue raised in the chromium bugtracker: https://bugs.chromium.org/p/chromedriver/issues/detail?id=2307

**Comment by aasimali** [comment]
@pulkitsharma07 

> Files can now be downloaded in headless mode after setting the download directory's path using the #download_path= helper.

**How Can I do that? Can you please tell me the steps? I use Ubuntu**

**Comment by p0deje** [comment]
I don't know how I missed that PR at all 😕 The proposed changes were actually implemented in f77d2f9e60, so users can use `driver.download_path = '/home`.

Sorry if this PR was a waste of your time, I somehow missed it completely. Is there anything else we need to take from here

**Comment by SergeyPirogov** [comment]
Hello, maybe someone know how to send DevTools command?

I am able to use

```
browser.defineCustomCommand("sendCommand", new CommandInfo("/session/:sessionId/goog/cdp/execute", POST));

Object result = browser.sendCommand("Debugger.enable", new HashMap<>());
````

but I can not subscribe to event:
```
Object execute = browser.sendCommand("Debugger.scriptParsed", new HashMap<>());
```

**Pull request closed without merging**